examples/deployment/docker-compose.yml

```diff
@@ -21,10 +25,19 @@ services:
       - "8090:8090"
       - "8091:8091"
     depends_on:
-      - mysql
+      - db
     environment:
       - DB_USER=root
       - DB_PASSWORD=$DB_PASSWORD
+    # can't use service's `environment` variables in `command`
+    # could use environment variables from the shell (!) in `command`
```

Comment:
https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/#use-environment-variables-to-define-arguments suggests otherwise.

Reply:
Kubernetes supports this -- as you reference -- but this is not possible in Docker Compose if the command is not a shell; with a specific process there's not shell (pre-processing) of the environment variables and Docker Compose provides no specific mechanism to do this otherwise.

The workaround in Docker Compose is to reference these values when `docker-compose` is invoked:
`docker-compose run -e DB_PASSWORD=...`

Reply:
Ah of course, I'm so used to looking at Kubernetes configs that I didn't realise I was looking at a Docker Compose config instead.